File: src/main/java/org/funivan/lologa/tiles/Filter/PositionFilter.java
package org.funivan.lologa.tiles.Filter;

import org.funivan.lologa.tile.Position.PositionInterface;
import org.funivan.lologa.tile.TileInterface;

public class PositionFilter implements TileFilterInterface {

    private final PositionInterface position;
    private final TileFilterInterface.EXPECT expect;

    public PositionFilter(PositionInterface positions, EXPECT expect) {
        this.position = positions;
        this.expect = expect;
    }

    public PositionFilter(PositionInterface positions) {
        this(positions, EXPECT.Positive);
    }


    @Override
    final public Boolean apply(TileInterface target) throws Exception {
        final boolean expectResult = this.expect == EXPECT.Positive;
        final boolean result = this.position.equals(target.position());
        return expectResult == result;
    }
}

File: src/main/java/org/funivan/lologa/tiles/filter/PositionFilter.kt
package org.funivan.lologa.tiles.filter

import org.funivan.lologa.tile.Position.PositionInterface
import org.funivan.lologa.tile.TileInterface

class PositionFilter(
        private val position: PositionInterface, private val expect: TileFilterInterface.EXPECT = TileFilterInterface.EXPECT.Positive
) : TileFilterInterface {

    override fun apply(target: TileInterface): Boolean {
        val expectResult = this.expect == TileFilterInterface.EXPECT.Positive
        val result = this.position == target.position()
        return expectResult == result
    }
}
